src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -93,27 +93,39 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 	}
 	
 	@Override
-	public Collection<AddOnInfoSummary> search(AddOnType type, String query) throws IOException {
+	public Collection<AddOnInfoSummary> search(AddOnType type, String query, String tag) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
-			//Gets all modules of the type specified by variable type
+			//Exact match on type
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
+		if (tag != null) {
+			
+		SearchResult result = client.execute(new Search.Builder(new SearchSourceBuilder()
+				.size(SEARCH_SIZE)
+				.query(QueryBuilders.matchQuery("tags", tag)).toString())
+				.addIndex(AddOnInfoAndVersions.ES_INDEX)
+				.build());
+		return result.getHits(AddOnInfoAndVersions.class).stream()
```

Comment:
So, the logic you've implemented here means that it will _only_ search by tag. And this would be confusing for someone who wants to call this method, if you're allowed to search for type and query optionally together, or tag alone.

But I think it's trivial to support searching by whatever combination of criteria the caller passes in. You'd just add to boolQB like is done in the if/then block about 'type'